libraries/joomla/database/driver/mysqli.php

```
@@ -557,8 +557,7 @@ public function execute()
 
 		if (!is_object($this->connection))
 		{
-			JLog::add(JText::sprintf('JLIB_DATABASE_QUERY_FAILED', $this->errorNum, $this->errorMsg), JLog::ERROR, 'database');
-			throw new RuntimeException($this->errorMsg, $this->errorNum);
+			throw new RuntimeException('Could not connect to MySQL.');
```

I think you're right, it should throw an Exception